sdk/__init__.py

```diff
@@ -48,19 +55,26 @@ def create_batch(
         Args:
             serialized_sequence: Serialized pulser sequence.
             jobs: List of jobs to be added to the batch at creation. (#TODO: Make optional after Iroise MVP)
-            emulator: Whether to run the batch on an emulator.
-              If set to false, the device_type will be set to the one
+            device_type: The type of device to use, either an emulator or a QPU
+              If set to QPU, the device_type will be set to the one
               stored in the serialized sequence
             wait: Whether to wait for results to be sent back
 
 
         Returns:
             Batch: The new batch that has been created in the database.
         """
+
+        emulator = (
+            device_type == DeviceType.CPU_EMULATOR
+            or device_type == DeviceType.GPU_EMULATOR
+        )
+
         batch_rsp, jobs_rsp = self._client._send_batch(
             {
                 "sequence_builder": serialized_sequence,
                 "emulator": emulator,
+                "device_type": device_type,
```

Comment:
I feel like havng both these "emulator" and "device_type" fields is a bit redundant. I would rather keep only the "emulator" field and replace the boolean value by an StrEnum value 

On the API side we would check whether the emulator field is in the request body: 
- if yes, set the device_type to that value (insteadof hardcoded emulator right now)
- if not extract the device_type from the pulser sequence (already done)

Thoughts ?

Reply:
@MatthieuMoreau0 Yes, that's a good idea. It might also lead to slight simplification of the code in the backend.